x/coin.go

```diff
@@ -51,6 +51,40 @@ func (c Coin) ID() string {
 	return c.Issuer + "/" + c.Ticker
 }
 
+// Split divides the value of a coin into given amount of pieces and returns a
+// single piece.
+// It might be that a precise splitting is not possible. Any leftover of a
+// fractional value is returned as well.
+// For example splitting 4 EUR into 3 pieces will result in a single piece
+// being 1.33 EUR and 1 cent returned as the rest (leftover).
+//   4 = 1.33 x 3 + 1
+func (c Coin) Divide(pieces int64) (one, rest Coin) {
+	// This is an invalid use of the method.
```

Comment:
yeah, is zero, zero the best error message?
I feel adding a third "error" argument is annoying, as it should never happen except on illegal pieces value (it doesn't depend on c)

Reply:
I can `panic` as an alternative. I think that returning an error is the worse solution because it makes the API very verbose.

Reply:
pp: "zero, zero" feels wrong. And I don't like panics either :-) 
I would prefer adding error or a ResultObject, with all 3 value. Would an ErrorCoin make sense?

Reply:
If we could move `Divide` and `Multiply` func into a new BigCoin type for high precision math, we can also have an error state internally and check on conversion. For example:
````
bigResult, rest := NewBigCoin(oldCoin).Divide(pieces)
coinType, err := bigResult.AsCoin()
```

Reply:
This is continued in https://github.com/iov-one/weave/pull/322#discussion_r257604214

Reply:
The decision was `BigCoin` is "outside of scope of this PR" by @ethanfrey. 
I vote for returning the error. This is consistent with what we do on add/sub .